File: test/fakers/category_faker.dart
import 'package:faker/faker.dart';
import 'package:sertton/core/catalog/dtos/category_dto.dart';

class CategoryFaker {
  static final faker = Faker();

  static CategoryDto fakeDto({String? id, String? name, String? description}) {
    return CategoryDto(
      id: id ?? faker.guid.guid(),
      name: name ?? faker.lorem.words(2).join(' '),
      description: description ?? faker.lorem.sentence(),
    );
  }

  static List<CategoryDto> fakeManyDto({
    int count = 10,
    String? id,
    String? name,
    String? description,
  }) {
    return List.generate(
      count,
      (index) => fakeDto(id: id, name: name, description: description),
    );
  }
}

File: sertton-react-native-app/src/core/catalog/entities/fakers/category-faker.ts
import { faker } from "@faker-js/faker"

import type { Category } from "../category"

export class CategoryFaker {
  static fake(category?: Partial<Category>): Category {
    return {
      description: faker.lorem.sentence(),
      id: faker.string.uuid(),
      name: faker.commerce.department(),
      ...category,
    }
  }

  static fakeMany(count = 10, category?: Partial<Category>): Category[] {
    return Array.from({ length: count }, () => CategoryFaker.fake(category))
  }
}
